doc/content/system/scheduler.md

```diff
@@ -1,4 +1,68 @@
+
+The [SimpleMPIScheduler](#simplempischeduler) is an extension of the [SimpleScheduler](#simplescheduler) which better distributes work in MPI parallel applications calling NEML2.  This scheduler assigns each MPI rank a unique accelerator device out of a list of devices provided by the user.  This MPI rank sends work to its assigned device exclusively, using the same algorithm for work scheduling as in the [SimpleScheduler](#simplescheduler).  The difference in inputs between the [SimpleMPIScheduler](#simplempischeduler) and the [SimpleScheduler](#simplescheduler) is just that the user now provides a list of devices, a list of subbatch sizes, and (optionally), a list of capacities.  Each item in the list is a single device available to MPI ranks.
+
+This scheduler assigns a rank to a device by first forming a local MPI communicator over the MPI ranks running on a single compute node (determined by the hostname provided by the [hwloc library](https://www.open-mpi.org/projects/hwloc/)).  Each MPI rank in the communicator is assigned one device from the list provided by the user.  The user must provide at least as many devices as the local communicator has MPI ranks.  This approach limits the number of MPI ranks a user can launch *per node* to the number of available accelerators.  For example, if a compute node has one CPU with 16 threads and 8 GPU accelerators the host application can only run 8 MPI ranks per node.  The scheduler will report an error if the host application tries to run it with more local ranks than user supplied devices.  This often means the user will need to [pin ranks](https://hpc-wiki.info/hpc/Binding/Pinning) to ensure their MPI application assigns only the required number of ranks per node.
+
+This scheduler is useful for larger parallel jobs, but may limit the overall performance of the application as often there are more CPU cores available for MPI ranks than their are accelerators on a node.
+
+### StaticHybridScheduler
+
+The [StaticHybridScheduler](#statichybridscheduler) allows a single instance of NEML2 to distribute work to many compute devices at once.  The user provides a list of devices, subbatch sizes, capacities, and (optionally) priorities.  The scheduler will greedily assign work to fill up each device, using the priority if provided to determine in which order to dole out work.  The work dispatcher will then run this work in parallel on all devices via the thread pool discussed above.
+
+This scheduler currently provides the best throughput for cases where a single NEML2 instance has access to multiple devices.  Unlike the simple schedulers, the device list can include the CPU itself so that the CPU thread running NEML2 can do work at the same time the accelerators are working on their subbatches.  We suggest the user determine the optimal subbatch size for each device, for example using the [SimpleScheduler](#simplescheduler), prior to running large jobs with this scheduling algorithm.
```

Comment:
> Unlike the simple schedulers, the device list can include the CPU itself so that the CPU thread running NEML2 can do work at the same time the accelerators are working on their subbatches.

Similar question as before -- I thought the simple scheduler can use CPU.